.travis.yml

```diff
@@ -33,10 +33,20 @@ jobs:
     install:
     script: ./scripts/ci/check_patch.sh
   # test jobs
+  - name: "3.8 on Windows"
```

Comment:
2 tests on Windows + Python3.8 are failing.

<details>
<summary>test_cache_type_is_properly_overridden</summary>

####
```
___________________ test_cache_type_is_properly_overridden ____________________
[gw3] win32 -- Python 3.8.0 c:\python38\python.exe
self = ExternalRepo: 'C:\Users\travis\AppData\Local\Temp\tmp1b0s1wjldvc-erepo'
path = 'foo', to_info = WindowsPathInfo: 'foo_imported'
    def pull_to(self, path, to_info):
        """
        Pull the corresponding file or directory specified by `path` and
        checkout it into `to_info`.
    
        It works with files tracked by Git and DVC, and also local files
        outside the repository.
        """
        out = None
        path_info = PathInfo(self.root_dir) / path
    
        with suppress(OutputNotFoundError):
            (out,) = self.find_outs_by_path(fspath(path_info), strict=False)
    
        try:
            if out and out.use_cache:
>               self._pull_cached(out, path_info, to_info)
C:\Users\travis\build\iterative\dvc\dvc\external_repo.py:87: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
self = ExternalRepo: 'C:\Users\travis\AppData\Local\Temp\tmp1b0s1wjldvc-erepo'
out = OutputLOCAL: 'foo'
path_info = WindowsPathInfo: '..\..\..\tmp1b0s1wjldvc-erepo\foo'
dest = WindowsPathInfo: 'foo_imported'
    def _pull_cached(self, out, path_info, dest):
        with self.state:
            tmp = PathInfo(tmp_fname(dest))
            src = tmp / path_info.relative_to(out.path_info)
    
            out.path_info = tmp
    
            # Only pull unless all needed cache is present
            if out.changed_cache(filter_info=src):
                self.cloud.pull(out.get_used_cache(filter_info=src))
    
            failed = out.checkout(filter_info=src)
    
>           move(src, dest)
C:\Users\travis\build\iterative\dvc\dvc\external_repo.py:111: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
src = 'C:\\Users\\travis\\AppData\\Local\\Temp\\pytest-of-travis\\pytest-0\\test_cache_type_is_properly_ov1\\foo_imported.kg2iDAmXzShzSpJf2jCxJx.tmp'
dst = 'C:\\Users\\travis\\AppData\\Local\\Temp\\pytest-of-travis\\pytest-0\\test_cache_type_is_properly_ov1\\foo_imported'
mode = None
    def move(src, dst, mode=None):
        """Atomically move src to dst and chmod it with mode.
    
        Moving is performed in two stages to make the whole operation atomic in
        case src and dst are on different filesystems and actual physical copying
        of data is happening.
        """
    
        src = fspath(src)
        dst = fspath(dst)
    
        dst = os.path.abspath(dst)
        tmp = "{}.{}".format(dst, uuid())
    
        if os.path.islink(src):
>           shutil.copy(os.readlink(src), tmp)
C:\Users\travis\build\iterative\dvc\dvc\utils\fs.py:107: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
src = '\\\\?\\C:\\Users\\travis\\AppData\\Local\\Temp\\pytest-of-travis\\pytest-0\\test_cache_type_is_properly_ov1\\.dvc\\cache\\6d\\bda444875c24ec1bbdb433456be11f'
dst = 'C:\\Users\\travis\\AppData\\Local\\Temp\\pytest-of-travis\\pytest-0\\test_cache_type_is_properly_ov1\\foo_imported.G8faL7XRJsyMfDFiQq6hF3'
    def copy(src, dst, *, follow_symlinks=True):
        """Copy data and mode bits ("cp src dst"). Return the file's destination.
    
        The destination may be a directory.
    
        If follow_symlinks is false, symlinks won't be followed. This
        resembles GNU's "cp -P src dst".
    
        If source and destination are the same file, a SameFileError will be
        raised.
    
        """
        if os.path.isdir(dst):
            dst = os.path.join(dst, os.path.basename(src))
>       copyfile(src, dst, follow_symlinks=follow_symlinks)
c:\python38\lib\shutil.py:409: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
src = '\\\\?\\C:\\Users\\travis\\AppData\\Local\\Temp\\pytest-of-travis\\pytest-0\\test_cache_type_is_properly_ov1\\.dvc\\cache\\6d\\bda444875c24ec1bbdb433456be11f'
dst = 'C:\\Users\\travis\\AppData\\Local\\Temp\\pytest-of-travis\\pytest-0\\test_cache_type_is_properly_ov1\\foo_imported.G8faL7XRJsyMfDFiQq6hF3'
follow_symlinks = True
    def copyfile(src, dst, follow_symlinks=True):
        """Copy data from src to dst.
        It uses windows native CopyFileW method to do so, making advantage of
        server-side copy where available.
        """
    
        if shutil._samefile(src, dst):
            # Get shutil.SameFileError if available (Python 3.4+)
            # else fall back to original behavior using shutil.Error
            SameFileError = getattr(shutil, "SameFileError", shutil.Error)
            raise SameFileError(
                "{!r} and {!r} are the same file".format(src, dst))
    
        for fn in [src, dst]:
            try:
                st = os.stat(fn)
            except OSError:
                # File most likely does not exist
                pass
            else:
                # XXX What about other special files? (sockets, devices...)
                if stat.S_ISFIFO(st.st_mode):
                    raise shutil.SpecialFileError("`%s` is a named pipe" % fn)
    
        if not follow_symlinks and os.path.islink(src):
            os.symlink(os.readlink(src), dst)
        else:
            kernel32 = ctypes.WinDLL('kernel32',
                                     use_last_error=True, use_errno=True)
            copyfile = kernel32.CopyFile2
            copyfile.argtypes = (ctypes.c_wchar_p,
                                 ctypes.c_wchar_p,
                                 ctypes.c_void_p)
            copyfile.restype = ctypes.HRESULT
    
            source_file = os.path.abspath(os.path.normpath(src))
            dest_file = os.path.abspath(os.path.normpath(dst))
            if source_file.startswith('\\\\'):
                source_file = 'UNC\\' + source_file[2:]
            if dest_file.startswith('\\\\'):
                dest_file = 'UNC\\' + dest_file[2:]
    
>           ret = copyfile('\\\\?\\' + source_file,
                           '\\\\?\\' + dest_file, None)
c:\python38\lib\site-packages\speedcopy\__init__.py:228: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
>   ???
E   FileNotFoundError: [WinError -2147024843] The network path was not found
_ctypes/callproc.c:930: FileNotFoundError
During handling of the above exception, another exception occurred:
tmp_dir = WindowsTmpDir('C:/Users/travis/AppData/Local/Temp/pytest-of-travis/pytest-0/test_cache_type_is_properly_ov1')
scm = Git: 'C:\Users\travis\AppData\Local\Temp\pytest-of-travis\pytest-0\test_cache_type_is_properly_ov1\.git'
dvc = Repo: 'C:\Users\travis\AppData\Local\Temp\pytest-of-travis\pytest-0\test_cache_type_is_properly_ov1'
erepo_dir = WindowsTmpDir('C:/Users/travis/AppData/Local/Temp/pytest-of-travis/pytest-0/erepo15')
    def test_cache_type_is_properly_overridden(tmp_dir, scm, dvc, erepo_dir):
        with erepo_dir.chdir():
            with erepo_dir.dvc.config.edit() as conf:
                conf["cache"]["type"] = "symlink"
            erepo_dir.dvc.cache = Cache(erepo_dir.dvc)
            erepo_dir.scm_add(
                [erepo_dir.dvc.config.files["repo"]],
                "set source repo cache type to symlink",
            )
        assert System.is_symlink(erepo_dir / "foo")
    
>       dvc.imp(fspath(erepo_dir), "foo", "foo_imported")
C:\Users\travis\build\iterative\dvc\tests\func\test_import.py:210: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
C:\Users\travis\build\iterative\dvc\dvc\repo\imp.py:6: in imp
    return self.imp_url(path, out=out, erepo=erepo, locked=True)
C:\Users\travis\build\iterative\dvc\dvc\repo\__init__.py:27: in wrapper
    ret = f(repo, *args, **kwargs)
C:\Users\travis\build\iterative\dvc\dvc\repo\scm_context.py:4: in run
    result = method(repo, *args, **kw)
C:\Users\travis\build\iterative\dvc\dvc\repo\imp_url.py:28: in imp_url
    stage.run()
c:\python38\lib\site-packages\funcy\decorators.py:39: in wrapper
    return deco(call, *dargs, **dkwargs)
C:\Users\travis\build\iterative\dvc\dvc\stage.py:161: in rwlocked
    return call()
c:\python38\lib\site-packages\funcy\decorators.py:60: in __call__
    return self._func(*self._args, **self._kwargs)
C:\Users\travis\build\iterative\dvc\dvc\stage.py:941: in run
    self.deps[0].download(self.outs[0])
C:\Users\travis\build\iterative\dvc\dvc\dependency\repo.py:81: in download
    repo.pull_to(self.def_path, to.path_info)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
self = ExternalRepo: 'C:\Users\travis\AppData\Local\Temp\tmp1b0s1wjldvc-erepo'
path = 'foo', to_info = WindowsPathInfo: 'foo_imported'
    def pull_to(self, path, to_info):
        """
        Pull the corresponding file or directory specified by `path` and
        checkout it into `to_info`.
    
        It works with files tracked by Git and DVC, and also local files
        outside the repository.
        """
        out = None
        path_info = PathInfo(self.root_dir) / path
    
        with suppress(OutputNotFoundError):
            (out,) = self.find_outs_by_path(fspath(path_info), strict=False)
    
        try:
            if out and out.use_cache:
                self._pull_cached(out, path_info, to_info)
                return
    
            # Check if it is handled by Git (it can't have an absolute path)
            if os.path.isabs(path):
                raise FileNotFoundError
    
            fs_copy(fspath(path_info), fspath(to_info))
        except FileNotFoundError:
>           raise PathMissingError(path, self.url)
E           dvc.exceptions.PathMissingError: The path 'foo' does not exist in the target repository 'C:\Users\travis\AppData\Local\Temp\pytest-of-travis\pytest-0\erepo15' neither as an output nor a git-handled file.

```
</details>


<details>
<summary>TestCheckoutMovedCacheDirWithSymlinks.test</summary>

####
```
_________________ TestCheckoutMovedCacheDirWithSymlinks.test __________________
[gw2] win32 -- Python 3.8.0 c:\python38\python.exe
self = <tests.func.test_checkout.TestCheckoutMovedCacheDirWithSymlinks testMethod=test>
    def test(self):
        ret = main(["config", "cache.type", "symlink"])
        self.assertEqual(ret, 0)
    
        ret = main(["add", self.FOO])
        self.assertEqual(ret, 0)
    
        ret = main(["add", self.DATA_DIR])
        self.assertEqual(ret, 0)
    
        self.assertTrue(System.is_symlink(self.FOO))
        old_foo_link = os.readlink(self.FOO)
    
        self.assertTrue(System.is_symlink(self.DATA))
        old_data_link = os.readlink(self.DATA)
    
        old_cache_dir = self.dvc.cache.local.cache_dir
        new_cache_dir = old_cache_dir + "_new"
        os.rename(old_cache_dir, new_cache_dir)
    
        ret = main(["cache", "dir", new_cache_dir])
        self.assertEqual(ret, 0)
    
        ret = main(["checkout", "-f"])
        self.assertEqual(ret, 0)
    
        self.assertTrue(System.is_symlink(self.FOO))
        new_foo_link = os.readlink(self.FOO)
    
        self.assertTrue(System.is_symlink(self.DATA))
        new_data_link = os.readlink(self.DATA)
    
>       self.assertEqual(
            relpath(old_foo_link, old_cache_dir),
            relpath(new_foo_link, new_cache_dir),
        )
E       AssertionError: '\\\\[80 chars]K73avCdBf\\.dvc\\cache\\ac\\bd18db4cc2f85cedef654fccc4a4d8' != '\\\\[80 chars]K73avCdBf\\.dvc\\cache_new\\ac\\bd18db4cc2f85cedef654fccc4a4d8'
E       - \\?\C:\Users\travis\AppData\Local\Temp\dvc-test.4308.8uq67ygu.44B3hc3sGRapiK73avCdBf\.dvc\cache\ac\bd18db4cc2f85cedef654fccc4a4d8
E       + \\?\C:\Users\travis\AppData\Local\Temp\dvc-test.4308.8uq67ygu.44B3hc3sGRapiK73avCdBf\.dvc\cache_new\ac\bd18db4cc2f85cedef654fccc4a4d8
E       ?                                                                                                ++++
..\..\..\..\build\iterative\dvc\tests\func\test_checkout.py:460: AssertionError
---------------------------- Captured stdout call -----------------------------
2020-03-16 02:43:06,067 DEBUG: Writing 'C:\Users\travis\AppData\Local\Temp\dvc-test.4308.8uq67ygu.44B3hc3sGRapiK73avCdBf\.dvc\config'.
```
</details>

Reply:
Trying without optimization of `speedcopy`.